scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -0,0 +1,1122 @@
+
+    @property
+    def seed(self):
+        return self.numpy_rng
+
+    @seed.setter
+    def seed(self, new_seed):
+        self.numpy_rng = get_numpy_rng(new_seed)
+        _lock.acquire()
+        try:
+            self._messages.clear()
+            unur_set_stream(self._messages.handle)
+            unur_urng_free(self.urng)
+            self._urng_builder = _URNG(self.numpy_rng)
+            self.urng = self._urng_builder.get_urng()
+            if self.urng == NULL:
+                raise UNURANError(self._messages.get())
+            unur_chg_urng(self.rng, self.urng)
+        finally:
+            _lock.release()
+
+    @cython.final
+    def __dealloc__(self):
+        if self.distr != NULL:
+            unur_distr_free(self.distr)
+            self.distr = NULL
+        if self.par != NULL:
+            unur_par_free(self.par)
+            self.par = NULL
+        if self.rng != NULL:
+            unur_free(self.rng)
+            self.rng = NULL
+        if self.urng != NULL:
+            unur_urng_free(self.urng)
+            self.urng = NULL
+
+    # Pickling support
+    @cython.final
+    def __reduce__(self):
+        klass = functools.partial(self.__class__, **self._kwargs)
+        return (klass, ())
+
+
+cdef class TransformedDensityRejection(Method):
+    r"""
+    Transformed Density Rejection (TDR) Method.
+
+    TDR is an acceptance/rejection method that uses the concavity of a
+    transformed density to construct hat function and squeezes automatically.
+    Most universal algorithms are very slow compared to algorithms that are
+    specialized to that distribution. Algorithms that are fast have a slow
+    setup and require large tables. The aim of this universal method is to
+    provide an algorithm that is not too slow and needs only a short setup.
+    This method can be applied to univariate and unimodal continuous
+    distributions with T-concave density function. See [1]_ and [2]_ for
+    more details.
+
+    Parameters
+    ----------
+    dist : object
+        An instance of a class with 'pdf' and 'dpdf' methods.
+
+        * 'pdf': PDF of the distribution. The signature of the PDF is
+          expected to be: ``def pdf(x: float, *params) -> float``. i.e.
+          the PDF should accept a Python float and the parameters and
+          return a Python float. It doesn't need to integrate to 1 i.e.
+          the PDF doesn't need to be normalized.
+        * 'dpdf': Derivative of the PDF w.r.t x (i.e. the variate). Must
+          have the same signature as the PDF.
+
+    mode : float, optional
+        Mode of the distribution. It is optional and only required if
+        `use_mode` is ``True``.
+    center : float, optional
+        Center (``loc``) of the distribution. It is optional and only
+        required if `use_center` is ``True``. Default is 0.
+    params : tuple, optional
+        Parameters that the PDF and DPDF take. Default is an empty tuple.
```

Comment:
The intent of the two comments above is to simplify the interface. Continuing along those lines, I'd lightly suggest that we eliminate this parameter. If it is very important to provide this as a convenience parameter, why is not equally important to accept `kwds` to be passed in as keyword (-only) arguments? Rather than adding both, I'd suggest adding neither. It's not so bad here with only these two, but sometimes things like this start to get out of hand (e.g. see where gh-11911 [was headed](https://github.com/scipy/scipy/pull/11911#discussion_r415261720)).

Reply:
> If it is very important to provide this as a convenience parameter, why is not equally important to accept `kwds` to be passed in as keyword (-only) arguments?

You are right. IMO, there should ideally be a `kwargs` keyword to do that. I added this parameter mainly because many functions in SciPy have it (e.g. integrate.quad, many optimize functions) and I wanted to be consistent. But it doesn't have to be a precedent.

> Rather than adding both, I'd suggest adding neither. It's not so bad here with only these two, but sometimes things like this start to get out of hand

I agree. One can use `functools.partial` to get similar behavior. Though, should we wait for others to comment on this before I change it? I think it would be better to incorporate other's thoughts before making a final decision.

Reply:
Yes, go ahead and wait. 
`functools.partial` is probably the most specific tool for the job, but even if users aren't aware of that, they should know how to define a function (`lambda` or otherwise).
I agree it has been common in the past. But since it is left out of some other stats functions that accept a function as an argument, you can say there's also precedent for that : )(Of course I probably created those, but whatever... : P

Reply:
I am fine to remove `params`

Reply:
done.

Reply:
params has been removed, I am resolving this conversation